### PR TITLE
Fix: counting server restart times error

### PR DESCRIPTION
**Pull request by kerthcet**
restartCount starts with 0, before, we'll crash 7 times then quit. Now, we'll quit at 6th.

/kind bug

### CODE REVIEW

**Comment by kerthcet** [comment]
close in favor of https://gitlab.com/nvidia/kubernetes/device-plugin/-/merge_requests/344

**Comment by elezar** [comment]
@kerthcet sorry for the back and forth on this. We are in the process of migrating our repos to GitHub. Would it be possible to reopen this as a PR and close https://gitlab.com/nvidia/kubernetes/device-plugin/-/merge_requests/344 instead?

**Comment by kerthcet** [comment]
Glad to hear that. reopened.

File: internal/plugin/server.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ func (plugin *NvidiaDevicePlugin) Serve() error {
 
 			// restart if it has not been too often
 			// i.e. if server has crashed more than 5 times and it didn't last more than one hour each time
-			if restartCount > 5 {
+			if restartCount >= 5 {
 				// quit
 				klog.Fatalf("GRPC server for '%s' has repeatedly crashed recently. Quitting", plugin.rm.Resource())
 			}
```
Comment (elezar) [inline internal/plugin/server.go:180]: Let's either move this to the start or the end of the loop to make it more consistent. This should also allow us to keep `restartCount > 5` as the conditional.

File: internal/plugin/server.go [hunk 2]
```diff
@@ -183,14 +183,14 @@ func (plugin *NvidiaDevicePlugin) Serve() error {
 			if timeSinceLastCrash > 3600 {
 				// it has been one hour since the last crash.. reset the count
 				// to reflect on the frequency
-				restartCount = 1
+				restartCount = 0
```
Comment (elezar) [inline internal/plugin/server.go:186]: 👍

**Comment by kerthcet** [comment]
Done @elezar

File: internal/plugin/server.go (hunk 1 above)
Comment (kerthcet) [reply internal/plugin/server.go:180]: Done

**Review by elezar — APPROVED** [review]

**Pull request merged**